// @wix/wix-data-items-common/dist/cjs/api/expressions/expressions.js
"use strict";

exports.__esModule = true;
exports.expressions = void 0;
var _Expression = require("./Expression");
const expressions = exports.expressions = {
  field: fieldPath => new _Expression.FieldExpressionImpl(fieldPath),
  text: value => new _Expression.TextExpressionImpl(value),
  numeric: value => new _Expression.NumericExpressionImpl(value),
  add: (...expressions) => new _Expression.AddExpressionImpl(expressions),
  subtract: (first, second) => new _Expression.SubtractExpressionImpl(first, second),
  multiply: (...expressions) => new _Expression.MultiplyExpressionImpl(expressions),
  divide: (first, second) => new _Expression.DivideExpressionImpl(first, second),
  sum: (...expressions) => new _Expression.SumExpressionImpl(expressions),
  abs: expression => new _Expression.AbsExpressionImpl(expression),
  mod: (first, second) => new _Expression.ModExpressionImpl(first, second),
  floor: expression => new _Expression.FloorExpressionImpl(expression),
  ceil: expression => new _Expression.CeilExpressionImpl(expression),
  concat: (...expressions) => new _Expression.ConcatExpressionImpl(expressions),
  stringify: expression => new _Expression.StringifyExpressionImpl(expression),
  toLower: expression => new _Expression.ToLowerExpressionImpl(expression),
  toUpper: expression => new _Expression.ToUpperExpressionImpl(expression),
  substring: (expression, start, length) => new _Expression.SubstringExpressionImpl(expression, start, length),
  length: expression => new _Expression.LengthExpressionImpl(expression)
};

/**
 * @builder
 */
