// @wix/wix-data-items-common/dist/cjs/api/stages/ProjectStage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
exports.__esModule = true;
exports.ProjectStageImpl = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
/**
 * @builder
 */

class ProjectStageImpl {
  constructor() {
    (0, _defineProperty2.default)(this, "fields", []);
  }
  include(...fieldNames) {
    fieldNames.forEach(fieldName => {
      this.fields.push({
        include: {
          fieldName
        }
      });
    });
    return this;
  }
  exclude(...fieldNames) {
    fieldNames.forEach(fieldName => {
      this.fields.push({
        exclude: {
          fieldName
        }
      });
    });
    return this;
  }
  reshape(expression, fieldName) {
    this.addField(expression, fieldName);
    return this;
  }
  nested(stage, fieldName) {
    return this.addNestedProjectionField(fieldName, stage.fields);
  }
  addField(expression, resultFieldName) {
    this.fields.push({
      reshape: {
        resultFieldName,
        expression: expression.toProto()
      }
    });
    return this;
  }
  addNestedProjectionField(resultFieldName, fields) {
    this.fields.push({
      projectNested: {
        resultFieldName,
        fields
      }
    });
    return this;
  }
  toProto() {
    return {
      projection: {
        fields: this.fields
      }
    };
  }
}
exports.ProjectStageImpl = ProjectStageImpl;
