// @wix/wix-data-items-common/dist/cjs/api/stages/stages.js
"use strict";

exports.__esModule = true;
exports.stages = void 0;
var _FilterStage = require("./FilterStage");
var _ProjectStage = require("./ProjectStage");
var _GroupStage = require("./GroupStage");
var _UnwindStage = require("./UnwindStage");
var _SortStage = require("./SortStage");
var _LimitStage = require("./LimitStage");
var _SkipStage = require("./SkipStage");
var _ObjectToArrayStage = require("./ObjectToArrayStage");
const stages = exports.stages = {
  filter: () => new _FilterStage.FilterStageImpl(),
  project: () => new _ProjectStage.ProjectStageImpl(),
  group: () => new _GroupStage.GroupStageImpl(),
  unwind: arrayFieldName => new _UnwindStage.UnwindStageImpl(arrayFieldName),
  sort: () => new _SortStage.SortStageImpl(),
  limit: count => new _LimitStage.LimitStageImpl(count),
  skip: count => new _SkipStage.SkipStageImpl(count),
  objectToArray: (objectFieldName, destinationFieldName) => new _ObjectToArrayStage.ObjectToArrayStageImpl(objectFieldName, destinationFieldName)
};

/**
 * @builder
 */
